packages/medusa/src/services/product.ts

```diff
@@ -956,32 +1015,38 @@ class ProductService extends BaseService {
     })
 
     return priceIndex > -1
-      ? await this.setAdditionalPrices(
-          product,
-          config.currency_code,
-          config.region_id,
-          config.cart_id,
-          config.customer_id,
-          config.include_discount_prices
-        )
+      ? await this.setAdditionalPrices(product, {
+          cart_id: config.cart_id,
+          currency_code: config.currency_code,
+          customer_id: config.customer_id,
+          include_discount_prices: config.include_discount_prices,
+          region_id: config.region_id,
+        })
       : product
   }
 
   /**
    * Creates a query object to be used for list queries.
-   * @param {object} selector - the selector to create the query from
-   * @param {object} config - the config to use for the query
-   * @return {object} an object containing the query, relations and free-text
+   * @param selector - the selector to create the query from
+   * @param config - the config to use for the query
+   * @return an object containing the query, relations and free-text
    *   search param.
    */
-  prepareListQuery_(selector, config) {
+  private prepareListQuery_(
```

Comment:
Even though the method pretty much "only" extracts q, I actually like having it in a separate method like `prepareListQuery_` because I think it helps reduce the noise from the methods where it's used like `listAndCount` and `list`. Personally, I'd much rather reading a method where `prepareListQuery_` is used and I just say "oh okay .. here we do something with the selector and config object" than start reading the code which extracts the "q" from the selector, then deletes it, etc. 😄

Reply:
should'n be part of the buildQuery utils? that way it already extract the q and everywhere it is extracting the q it would not be needed anymore.

Reply:
```ts
export type ExtendedFindConfig<TEntity> = FindConfig<TEntity> & {
  where: Partial<Writable<TEntity>>
  withDeleted?: boolean
  relations?: string[]
}
```
we could add `q` as part of that type

Reply:
Great idea! I gave this a shot, but unfortunately (and naturally), it broke the types in a few places where the `buildQuery` is used. I think it might be a tiny bit wiser to keep this PR a bit scoped and try to implement your suggestion in another PR 😄. Wdyt?

Reply:
Yeah sorry, you entirely right 👍